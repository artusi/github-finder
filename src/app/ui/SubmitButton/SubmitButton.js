import React from "react";
import PropTypes from "prop-types";
import styles from "./SubmitButton.module.css";

export function SubmitButton({
  id,
  children,
  onClick,
  customClassName,
  disabled
}) {
  return (
    <button
      id={id}
      className={`${styles.submitButton} ${customClassName}`}
      type="button"
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </button>
  );
}

SubmitButton.propTypes = {
  id: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
  customClassName: PropTypes.string,
  disabled: PropTypes.bool
};

SubmitButton.defaultProps = {
  customClassName: "",
  disabled: false
};

export default SubmitButton;
