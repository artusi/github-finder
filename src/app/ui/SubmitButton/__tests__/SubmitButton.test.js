import React from "react";
import { SubmitButton } from "../SubmitButton";

const initialProps = {
  id: "submitButton",
  onClick: jest.fn()
};

describe("<SubmitButton /> - render", () => {
  it("renders", () => {
    const wrapper = shallow(
      <SubmitButton {...initialProps}>Test submit</SubmitButton>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders with all props > customClassName | disabled", () => {
    const allProps = {
      ...initialProps,
      customClassName: "lol-my-class",
      disabled: true
    };
    const wrapper = shallow(
      <SubmitButton {...allProps}>Test submit</SubmitButton>
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("<SubmitButton /> - interactions", () => {
  it("Searching for something", () => {
    const simulateClick = jest.fn();

    const wrapper = shallow(
      <SubmitButton {...initialProps} onClick={simulateClick}>
        Test submit
      </SubmitButton>
    );

    wrapper.find("#submitButton").simulate("click");

    expect(simulateClick).toBeCalled();
  });
});
