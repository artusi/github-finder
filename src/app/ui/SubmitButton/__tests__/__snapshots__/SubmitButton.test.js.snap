// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton /> - render renders 1`] = `
<button
  className="submitButton "
  disabled={false}
  id="submitButton"
  onClick={[MockFunction]}
  type="button"
>
  Test submit
</button>
`;

exports[`<SubmitButton /> - render renders with all props > customClassName | disabled 1`] = `
<button
  className="submitButton lol-my-class"
  disabled={true}
  id="submitButton"
  onClick={[MockFunction]}
  type="button"
>
  Test submit
</button>
`;
